fem/bilininteg_vectorfe.cpp

```diff
@@ -243,8 +262,12 @@ void VectorFEMassIntegrator::AssembleDiagonalPA(Vector& diag)
    {
       if (fetype == mfem::FiniteElement::CURL)
       {
-         PAHcurlMassAssembleDiagonal3D(dofs1D, quad1D, ne,
-                                       mapsO->B, mapsC->B, pa_data, diag);
+         if (diag.GetMemory().GetMemoryType() >= MemoryType::DEVICE && quad1D <= 6)
```

Comment:
We should find a more convenient way to help with this test - we can add some call if you need.

Reply:
@camierjs I was also wondering about the `MemoryType::MANAGED` since that is what is checked in `IsDeviceMemory`. In other words, is it correct to use `mt >= MemoryType::MANAGED` or `mt >= MemoryType::DEVICE`?

Reply:
I agree that `MANAGED`, which is both host and device could be miss-used in such tests.
I'd prefer to use `Device::Allow(DEVICE_MASK)` to test if we are using a device.
So I'd answer: `mt >= MemoryType::DEVICE`.

Reply:
I don't think the decision which kernel to run should depend on the memory type. Specifically, `Memory::GetMemoryType()` returns the currently valid memory type, so you can get host memory type even when using a cuda backend. As @camierjs said, it is better to use `Device::Allows(Backend::DEVICE_MASK)` for the dispatch here.

Reply:
The condition suggested by @v-dobrev is being used now.